packages/client/hmi-client/src/utils/project-card.ts

```diff
@@ -0,0 +1,218 @@
+		p.noFill();
+		p.strokeWeight(lineNumber / number);
+		p.stroke(72 + randomShift, 20, 200, 0.6);
+
+		p.beginShape();
+		p.curveVertex(0, y - 2);
+		p.curveVertex(0, y - 2);
+		for (let i = 1; i < numPoints; i++) {
+			p.curveVertex((imageWidth / numPoints) * i, linePosition[i] - 2);
+		}
+		p.curveVertex(imageWidth, y - 2);
+		p.curveVertex(imageWidth, y - 2);
+		p.endShape();
+	}
+
+	for (let i = 1; i <= datasets; i++) {
+		const ySpacing = (imageHeight / datasets) * (i / 1.5);
+		squigglyLine(ySpacing, datasets, i);
+	}
+}
+
+// Cone-shaped slashes at random places across screen for each model
+function drawModels(p: p5) {
+	for (let i = 1; i <= models; i++) {
+		p.colorMode('hsb');
+		const strokeHue = p.random(180, 360);
+		const strokeFrom = p.color(strokeHue, 30, 70, 0.5);
+		const strokeTo = p.color(strokeHue, 50, 50, 0.5);
+		p.strokeWeight(2);
+		const topX = p.random(0, imageWidth);
+		const bottomX = p.random(0, imageWidth);
+		for (let j = 0; j <= 10; j++) {
+			p.stroke(p.lerpColor(strokeFrom, strokeTo, j / 10));
+			p.line(topX, 0, bottomX + j, imageHeight);
+		}
+	}
+}
+
+// Circles at random heights for each paper, decreasing in size as number increases
+function drawPapers(p: p5) {
+	function drawCircle(cx, cy, circleSize) {
+		p.noStroke();
+		p.colorMode('hsb');
+		const randomHue = p.random(50, 200);
+
+		// back
+		p.fill(randomHue, 100, 50);
+		p.circle(cx, cy, circleSize);
+
+		// feathered hot spot
+		for (let i = 1; i < 50; i++) {
+			p.fill(randomHue, 10, 100, 0.05);
+			p.ellipse(
+				cx - circleSize / 8,
+				cy - circleSize / 8,
+				circleSize * (i / 50),
+				circleSize * (i / 50)
+			);
+		}
+	}
+
+	for (let i = 1; i <= papers; i++) {
+		const xPos = p.map(i % 10, 0, 10, 20, imageWidth - 10, true);
+		const yPos = p.random(10, imageHeight - 20);
+		const circleSize = p.map(papers, 0, 50, 20, 10, true);
+		drawCircle(xPos, yPos, circleSize * p.random(1, 2));
+	}
+}
+
+// Setup the instance mode of p5
+// https://p5js.org/reference/#/p5/p5
+const sketch = (p: p5) => {
+	// Create the canvas
+	p.setup = () => {
+		p.createCanvas(imageWidth, imageHeight);
+		p.noCanvas(); // Do not display the canvas in the DOM
+		p.noLoop();
+		p.background(240);
+	};
+
+	p.draw = (): string => {
+		drawContributors(p);
+		if (models > 0) drawModels(p);
+		if (datasets > 0) drawDatasets(p);
+		if (papers > 0) drawPapers(p);
+		return (p as Ip5).canvas.toDataURL();
+	};
+};
+
+export function placeholder(args: IPlaceholderArgs): string {
+	// Set the value for each datum.
+	// Limit to 25 to keep the graph tidy.
+	contributors = Math.min(args.contributors, 25);
+	models = Math.min(args.models, 25);
+	datasets = Math.min(args.datasets, 25);
+	papers = Math.min(args.papers, 25);
+
+	// Create the p5 sketch
+	if (!placeholderCanvas) {
+		placeholderCanvas = new p5(sketch); /* eslint-disable-line new-cap */
+	}
+
+	// Add all of this within the canvas and export the base64 image
+	return placeholderCanvas.draw() as unknown as string;
```

Comment:
What does it mean to say `foobar as unknown as string` ?

Reply:
You can't transform a type to another directly, even if the kind of match, so you have to change it via unknown